src/components/WatchVideo.vue

```diff
@@ -648,6 +648,12 @@ export default {
             }
         },
         getVideoId() {
+            if (this.$route.query.video_ids) {
+                const videos_list = this.$route.query.video_ids.split(",");
+                this.index = Number(this.$route.query.index ?? 0);
+                return videos_list[this.index];
+            }
```

Comment:
This can be undefined if the index is past the video_ids size, we should do some bounds checking below.

Reply:
Good catch.